### ISSUE TITLE
bug: ion-alert opened with ion-select does not have isOpen set

### ISSUE THREAD

**Issue by ptmkenny** [ionic-team/ionic-framework#26902]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [ ] v6.x
- [X] v7.x
- [ ] Nightly

### Current Behavior

With an `ion-select`, clicking it opens an `ion-alert`.  Unlike a normal `ion-alert`, the `isOpen` attribute is not set, which makes it difficult to select.

### Expected Behavior

I'm writing Playwright tests and I want to get the currently open alert with a selector:

`ion-alert[is-open="true"]`

However, this doesn't work because `ion-alert` opened by `ion-select` does not have an `isOpen` attribute.

### Steps to Reproduce

Create an `ion-select` and click it.

See Tab 1 of the linked repo.

When you click the button to display an `ion-alert`, it has `isOpen` attribute set.

But when you click `ion-select`, the alert opens but `isOpen` attribute is not set.

### Code Reproduction URL

https://github.com/ptmkenny/ionic-26861/tree/alert

### Ionic Info

```
ionic info 
[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package'
       
       Require stack:
       - /opt/homebrew/lib/node_modules/@ionic/cli/lib/project/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/lib/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/bin/ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module '@capacitor/android/package'
       
       Require stack:
       - /opt/homebrew/lib/node_modules/@ionic/cli/lib/project/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/lib/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/index.js
       - /opt/homebrew/lib/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI       : 6.20.8 (/opt/homebrew/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 7.0.0-dev.11677608949.16e25264

Capacitor:

   Capacitor CLI      : 4.7.0
   @capacitor/android : not installed
   @capacitor/core    : 4.7.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.1

System:

   NodeJS : v19.6.0 (/opt/homebrew/Cellar/node/19.6.0/bin/node)
   npm    : 9.4.0
   OS     : macOS


```

**Comment by sean-perkins** [comment]
Hello @ptmkenny thanks for this issue.

The `is-open` property is only assigned for inline overlays. `ion-select` uses a controller internally to present the `ion-alert` overlay and will not apply `is-open` directly.

There are a few alternatives that may work for your scenario:
- Playwright has a [`:visible`](https://playwright.dev/docs/other-locators#css-matching-only-visible-elements) selector, which could be used with a locator for `page.locator('ion-alert:visible')`
- Ionic applies a `.select-alert` class to alerts presented from `ion-select`. You could have a locator for this selector, assert that it is visible and test whatever extra criteria you expect.
- Ionic has an auto-incrementing id that is applied to presented overlays, e.g.: `ion-overlay-1`. You could have your test query the overlay by its incrementing id.

Do any of these workarounds suffice for your use case?

**Comment by ptmkenny** [comment]
@sean-perkins Thanks for the detailed explanation.

The `:visible` selector pulls up old alerts that aren't visible as well, which is why I was looking to use `isOpen()`.  But those old alerts may be fixed by https://github.com/ionic-team/ionic-framework/issues/26185#issuecomment-1452926715, so as soon as I can get a build with that fix, I'll try it out and report back.

**Comment by sean-perkins** [comment]
Here is an updated dev-build of the `main` branch, which includes the commit for the React overlays:
```
npm i @ionic/react@6.6.1-dev.11677816488.1c0e1bf9 @ionic/react-router@6.6.1-dev.11677816488.1c0e1bf9
```

Let me know if that doesn't work and I can discuss with the team in more detail tomorrow. While we could assign `isOpen` to the overlay from the controller, it wouldn't be reflected to the DOM and be a queryable selector. We could make the attribute reflect when it is set, but we historically have only done this for styling and not for public API surface.

**Comment by ptmkenny** [comment]
@sean-perkins Thanks, I rolled back to a v6 version of my app and was able to get it working with the build linked above.

For the `ion-select` alert selector, I used `ion-app > ion-alert:visible`.

Note that because playwright executes tests so quickly, if you are selecting a value for one `ion-select` and then selecting another value for a different `ion-select` immediately after, and both `ion-selects` have an option with the same name, the test may see both names when using the above selector, causing a failure. (Even though only one ion-select picker should be visible at a time-- which is how the user sees it.) 

So in such cases, I had to put a delay in between steps when selecting:

`  await page.waitForTimeout(1000);
`

I'm going to go ahead and close this out but if you have any ideas about how to get playwright to wait for Ionic to finish processing before moving to the next step, I'd love to hear them. (There is `waitForSelector()` but the problem is that it's the same component being displayed multiple times, so `waitForSelector()` will pass with the component that is being removed, causing a test failure.)
